src/backend/distributed/utils/replication_origin_session_utils.c

```diff
@@ -0,0 +1,232 @@
+/*-------------------------------------------------------------------------
+ *
+ * replication_origin_session_utils.c
+ *   Functions for managing replication origin session.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "distributed/replication_origin_session_utils.h"
+#include "distributed/remote_commands.h"
+#include "distributed/metadata_cache.h"
+#include "utils/builtins.h"
+#include "miscadmin.h"
+
+static bool IsReplicationOriginSessionSetup(MultiConnection *connection);
+
+static bool ExecuteRemoteCommandAndCheckResult(MultiConnection *connection, char *command,
+											   char *expected);
+static RepOriginId originalOriginId = InvalidRepOriginId;
+
+/*static char* get_replication_origin_name(MultiConnection *connection); */
+
+static void repl_origin_local_session_start_no_publish(void);
+static void repl_origin_local_session_end_no_publish(void);
+static bool repl_origin_local_session_is_no_publish(void);
+
+PG_FUNCTION_INFO_V1(replication_origin_session_start_no_publish);
+PG_FUNCTION_INFO_V1(replication_origin_session_end_no_publish);
+PG_FUNCTION_INFO_V1(replication_origin_session_is_no_publish);
+
+Datum
+replication_origin_session_start_no_publish(PG_FUNCTION_ARGS)
+{
+	repl_origin_local_session_start_no_publish();
+	PG_RETURN_VOID();
+}
+
+
+Datum
+replication_origin_session_end_no_publish(PG_FUNCTION_ARGS)
+{
+	repl_origin_local_session_end_no_publish();
+	PG_RETURN_VOID();
+}
+
+
+Datum
+replication_origin_session_is_no_publish(PG_FUNCTION_ARGS)
+{
+	bool result = repl_origin_local_session_is_no_publish();
+	PG_RETURN_BOOL(result);
+}
+
+
+void
+repl_origin_local_session_start_no_publish(void)
```

Comment:
since these are not UDFs, it seems a bit nicer to stick to our usual function naming format

Reply:
Refactored the code to call SetupReplicationOriginRemoteSession and ResetReplicationOriginLocalSession functions which are anyway called from other places in the code.